iXBRLViewerPlugin/viewer/src/js/report.js

```diff
@@ -69,6 +69,34 @@ iXBRLReport.prototype._initialize = function () {
     }
 }
 
+iXBRLReport.prototype.isCalculationContributor = function(c) {
+    if (this._calculationContributors === undefined) {
+        if (this.data.rels?.calc) {
+            this._calculationContributors = new Set(Object.values(this.data.rels.calc).flatMap(calculations => {
+                return Object.values(calculations).flatMap(contributors => {
+                    return contributors.map(c => c.t);
+                });
+            }));
+        } else {
+            this._calculationContributors = [];
```

Comment:
Both here and in `isCalculationSummation`. The no calculations case should be an empty set. Otherwise the `has(...)` calls will throw type errors.